Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -476,12 +492,145 @@ void MemoryWidget::OnSearchAddress()
   m_search_offset->setPalette(offset_palette);
 }
 
-void MemoryWidget::ValidateSearchValue()
+void MemoryWidget::ValidateAndPreviewInputValue()
 {
+  m_data_preview->clear();
+  QString input_text = m_data_edit->text();
+  const auto combo_id = static_cast<InputId>(m_input_combo->currentData().toInt());
+
+  m_base_check->setEnabled(combo_id == InputId::U32 || combo_id == InputId::U16 ||
+                           combo_id == InputId::U8);
+
+  if (input_text.isEmpty())
+    return;
+
+  // Remove any spaces
+  if (combo_id != InputId::ASCII)
+    input_text.remove(QLatin1Char(' '));
+
   QFont font;
   QPalette palette;
+  QString hex_string;
+  bool good = true;
+
+  if (m_base_check->isChecked() && m_base_check->isEnabled())
+    input_text = QStringLiteral("0x") + input_text;
+
+  switch (combo_id)
+  {
+  case InputId::ASCII:
+  {
+    const QByteArray bytes = input_text.toLatin1();
+    hex_string = QString::fromLatin1(bytes.toHex());
+    break;
+  }
+  case InputId::FLOAT:
+  {
+    const float value_float = input_text.toFloat(&good);
+
+    if (good)
+    {
+      const u32 hex_out = Common::BitCast<u32>(value_float);
+      hex_string = QString::fromStdString(fmt::format("{:X}", hex_out));
+    }
+    break;
+  }
+  case InputId::DOUBLE:
+  {
+    const double value_double = input_text.toDouble(&good);
+
+    if (good)
+    {
+      const u64 hex_out = Common::BitCast<u64>(value_double);
+      hex_string = QString::fromStdString(fmt::format("{:X}", hex_out));
+    }
+    break;
+  }
+  case InputId::S8:
+  {
+    const short value = input_text.toShort(&good, 10);
+    if (good && value <= std::numeric_limits<signed char>::max() &&
```

Comment:
If the condition is more than one line, the `if` statement should be braced.

What about this, since `good` should be `false` by default.
```c++
const short value = input_text.toShort(&good, radix);
good &= std::numeric_limits<signed char>::min() <= value && value <= std::numeric_limits<signed char>::max();
if (good)
  hex_string = QString::fromStdString(fmt::sprintf("%02hhX", value));
break;
```